src/cache/cache_operations.rs

```diff
@@ -20,12 +25,64 @@ pub struct Invalidate {
     database: Database,
 }
 
+#[derive(Deserialize, Debug)]
+pub enum InvalidationScope {
+    Requests,
+    Responses,
+    Both,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(tag = "invalidate", content = "pattern_details")]
 pub enum InvalidationPattern {
-    FlushAll,
-    RequestsResponses(String),
+    Any(String, InvalidationScope),
+    Transactions(String, InvalidationScope),
+    Balances(String, InvalidationScope),
```

Comment:
Theoretically we also have transfers

Reply:
Yeah, I tried to find a regex that hits both the core and gw urls keys. That's the only thing I see as source of issues.